API-2 질문 추출 더미 데이터값으로 반환하는 엔드포인트 구축

silofn523 opened the pull request:
**작업 내용:**
재미나이 AI를 사용해서 텍스트를 보낼시에 질문을 추출하는 기능을 구현하기 위해서 만든,
**더미 데이터 값으로 미리 정보를 반환하는 API를 설계하고 엔드포인트를 구축했습니다.** 

**변경 사항 상세:** 
- 더비 데이터 반환하는 API 엔드포인트 구축
- 서비스 로직 구축 _(더미 데이터를 push해서 컨트롤러로 넘겨주는)_
- 인터셉터 메시지 반환부분 변경 _(컨트롤러에서 반환하는 메시지 값으로 변환)_

**주의 사항 & 중점적으로 봐야 할 부분:**
- 

**관련 이슈:**
없음

silofn523 committed:
Add files via upload

silofn523 committed:
라이브러리 다운

silofn523 committed:
app모듈 config모듈 중복 import제거

silofn523 committed:
API-2 질문 추출 더미 데이터값으로 엔드포인트 구축

oakkim commented on .env:
```diff
@@ -0,0 +1,14 @@
+SWAGGER_USER=user
+SWAGGER_PW=root
+
+GEMINI_API_KEY=AIzaSyCw9l1Bf77DMOMMe5YkUmhKzI9T0_4Rwds
```
Public 레포지토리인데, API Key 적어두면 해커가 채가요~ 깃헙에서 API 키 올린 경우 삭제하는 [방법](https://adjh54.tistory.com/382) 달아둘게.

(디비나 다른 비번은 테스트용이라 중요하지 않아서 상관없는데 API 키는 따로 잘 관리하는걸 추천)

oakkim commented on src/interceptors/json-request.interceptor.ts:
```diff
@@ -22,11 +17,16 @@ export class JsonRequestInterceptor implements NestInterceptor {
+        const responseData = data?.data !== undefined ? data.data : data
+
         return {
           success: true,
```
success 값을 돌려줄 필요가 있을까? HTTP Status Code가 200~299 사이라면 성공이라고 판단할 수 있어서

oakkim commented on src/interceptors/json-request.interceptor.ts:
```diff
@@ -22,11 +17,16 @@ export class JsonRequestInterceptor implements NestInterceptor {
           success: true,
-          message: '응답 성공',
+          message: message,
           statusCode,
```
Http Response Header 내에 statusCode가 이미 있는데, Body에 넣는게 의미는 없을 것 같아. 지워도 될 것 같아

oakkim commented on src/question/question.service.ts:
```diff
@@ -0,0 +1,36 @@
+    return newQuestion
+  }
+
+  findAll() {
+    return `This action returns all question`
+  }
+
+  findOne(id: number) {
+    return `This action returns a #${id} question`
+  }
+
+  update(id: number, updateQuestionDto: UpdateQuestionDto) {
+    return `This action updates a #${id} question`
+  }
+
+  remove(id: number) {
+    return `This action removes a #${id} question`
+  }
```
필요없는 코드는 삭제합시다!

oakkim commented on src/question/question.controller.ts:
```diff
@@ -0,0 +1,39 @@
+    }
+  }
+
+  @Get()
+  findAll() {
+    return this.questionService.findAll()
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.questionService.findOne(+id)
+  }
+
+  @Patch(':id')
+  update(@Param('id') id: string, @Body() updateQuestionDto: UpdateQuestionDto) {
+    return this.questionService.update(+id, updateQuestionDto)
+  }
+
+  @Delete(':id')
+  remove(@Param('id') id: string) {
+    return this.questionService.remove(+id)
+  }
```
필요없는 코드는 삭제하자!

oakkim commented on src/question/question.controller.ts:
```diff
@@ -0,0 +1,39 @@
+export class QuestionController {
+  constructor(private readonly questionService: QuestionService) {}
+
+  @Post('create-question')
```
`POST /question/create-question` 은 너무 중복된 네이밍 같네. `POST /questions/generate-by-text` 같은 Path도 좋을 것 같음.

그리고 `question`은 리소스 Dictionary를 뜻할테니, 복수형으로 사용하는게 좋을 것 같아.

oakkim commented on src/question/question.service.ts:
```diff
@@ -0,0 +1,36 @@
+
+@Injectable()
+export class QuestionService {
+  private question: [] = []
```
이 question 필드는 어떨때 쓰는 걸까? (create를 호출할 때마다 담아둘 이유가 따로 없을 거 같긴한데)

oakkim commented on src/question/question.service.ts:
```diff
@@ -0,0 +1,36 @@
+  private question: [] = []
+
+  public async create(dto: CreateQuestionDto) {
+    const newQuestion = {
```
newQuestion의 타입은 뭘까? create 함수의 리턴 타입은 뭘까?

oakkim commented on src/question/question.service.ts:
```diff
@@ -0,0 +1,36 @@
+    }
+    this.question.push(newQuestion as never)
+
+    return newQuestion
```
포트폴리오에 대한 질문을 생성하면 질문 하나만 나오지는 않을 것 같아.

oakkim reviewed (changes requested):
한번 체크해줘~

silofn523 committed:
Gemini-API로 텍스트에서 질문 추출
변경 사항
- Gemini-API연동하여 텍스트 기반으로 질문 추출하는 API개발
- 커스텀 예외처리 (Gemini err, DB err)
- DDD를 적용시켜 역할별 분리
- Redis를 이용하여 캐싱 (같은 텍스트 값으로 2번 이상 요청 보낼시)
- QuestionSet 엔티티에 사용자가 입력한 text값 저장, Question엔티티와 외래키로 연결 (어떤 텍스트를 기반으로 질문들이 생성됐는지 파악하기위해)
- Question 엔티티에 질문 리스트 저장

기타 참고 사항
- 관련 이슈: 2
